tests/test_common.py

```diff
@@ -42,6 +42,17 @@ def test_device(library):
     x2 = to_device(x, dev)
     assert device(x) == device(x2)
 
+@pytest.mark.parametrize("target_library,func", is_functions.items())
+@pytest.mark.parametrize("source_library", is_functions.keys())
+def test_asarray(source_library, target_library, func):
```

Comment:
+1 to this test. We are going to need to add a lot of tests here for asarray, partly because the test suite only handles things within the scope of the standard, and partly because the test suite itself doesn't test as much as it could (https://github.com/data-apis/array-api-tests/issues/241). We can also adopt the tests from https://github.com/data-apis/array-api-strict/pull/15 (once we have fixed the `copy` behavior, but I expect that will need to just be part of this overall fix since this is part of that logic).